integration_tests/data_utils.py

```diff
@@ -0,0 +1,199 @@
+# Code, lat_bounds, lon_bounds
+EPSG_CODES = [('EPSG:32755', (-80, 0), (144, 150))]
+crses = sampled_from(EPSG_CODES)
+REASONABLE_DATES = datetimes(min_value=datetime(1980, 1, 1), max_value=datetime(2030, 1, 1))
+
+
+@composite
+def bounds(draw, bounds):
+    return draw(
+        lists(
+            floats(*bounds), min_size=2, max_size=2
+        ).map(sorted).filter(lambda x: x[0] < x[1]))
+
+
+@composite
+def bboxes(draw, lat_bounds=(-90, 90), lon_bounds=(-180, 180)):
+    """
+    https://tools.ietf.org/html/rfc7946#section-5
+    """
+    # Use 3 dim positions even if we only need 2
+    min_lat, max_lat = draw(bounds(lat_bounds))
+    min_lon, max_lon = draw(bounds(lon_bounds))
+
+    return min_lon, min_lat, max_lon, max_lat
+
+
+@composite
+def extents(draw, lat_bounds, lon_bounds):
+    center_dt = draw(REASONABLE_DATES)
+
+    time_range = timedelta(seconds=12)
+    min_lon, min_lat, max_lon, max_lat = draw(bboxes(lat_bounds, lon_bounds))
+    return {
+        'center_dt': center_dt.strftime(DATE_FORMAT),
+        'from_dt': (center_dt - time_range).strftime(DATE_FORMAT),
+        'to_dt': (center_dt + time_range).strftime(DATE_FORMAT),
+        'coord': {
+            'll': {'lat': min_lat, 'lon': min_lon},
+            'lr': {'lat': min_lat, 'lon': max_lon},
+            'ul': {'lat': max_lat, 'lon': min_lon},
+            'ur': {'lat': max_lat, 'lon': max_lon},
+        }
+    }
+
+
+def _extent_point_projector(crs):
+    crs = CRS(crs)
+
+    def reproject_point(pos):
+        pos = point(pos['lon'], pos['lat'], CRS('EPSG:4326'))
+        coords = pos.to_crs(crs).coords[0]
+        return {'x': coords[0], 'y': coords[1]}
+
+    return reproject_point
+
+
+def extent_to_grid_spatial(extent, crs):
+    """Convert an extent in WGS84 to a grid spatial in the supplied CRS"""
+    reprojector = _extent_point_projector(crs)
+    return {
+        'projection': {
+            'geo_ref_points': {
+                corner: reprojector(pos)
+                for corner, pos in extent['coord'].items()
+            },
+            'spatial_reference': crs
+        }
+    }
+
+
+@composite
+def acquisition_details(draw):
+    return {
+        'aos': draw(REASONABLE_DATES),
+        'groundstation': {
+            'code': draw(text(alphabet=string.ascii_letters, min_size=3, max_size=5))
+        },
+        'los': draw(REASONABLE_DATES)
+    }
+
+
+@composite
+def scene_datasets(draw):
+    """
+    Generate random test Landsat 5 Scene Datasets
+    """
+    crs, lat_bounds, lon_bounds = draw(crses)
+    extent = draw(extents(lat_bounds, lon_bounds))
+    return {
+        'id': str(draw(uuids())),
+        'acquisition': draw(acquisition_details()),
+        'creation_dt': draw(REASONABLE_DATES),
+        'extent': extent,
+        'grid_spatial': extent_to_grid_spatial(extent, crs),
+        'image': {
+            'bands': {
+                '1': {
+                    'path': 'LS5_TM_NBAR_P54_GANBAR01-002_090_084_19900302_B10.tif'
```

Comment:
A bit concerned about this being fixed to a constant. Maybe right now ingest/index doesn't care, although if it goes into temp dir it might not matter?

Reply:
These are arbitrary file names inside of directories, so doesn't matter at the moment, but I'll replace it with randomly generated names.